### PR TITLE
fix: properly report workspace failed status

### PR DESCRIPTION
**Pull request by bennyz**
When the workspace pod fails to start we should report it via CLI and not get stuck on Creating

## Type of Change
- [X] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] CI/CD improvement
- [ ] Refactoring

## Testing
- [ ] Unit tests pass (`make test`)
- [ ] Linter passes (`make lint`)
- [ ] Manifests are up to date (`make manifests generate`)
- [ ] Tested on OpenShift cluster (if applicable)

## Summary by CodeRabbit

* **New Features**
  * Workspace status details now include server-provided reasons and messages.
  * Failed workspaces provide more specific troubleshooting guidance, including log commands for container and pod failures.
  * Timeout messages now include the latest workspace status and recommended next steps.

* **Bug Fixes**
  * Improved detection and reporting of container, pod, scheduling, storage, and image-related failures.
  * Workspace creation now rejects memory requests below 6 Mi with a clear validation error.

### CODE REVIEW

**Comment by bennyz** [comment]
@coderabbitai review

File: cmd/caib/workspace/workspace.go [hunk 1]
```diff
@@ -925,10 +933,15 @@ func waitForRunning(name string) {
...
-			))
+			err := fmt.Errorf("timed out waiting for workspace %q to be running", name)
+			if lastStatus != "" {
+				err = fmt.Errorf("%w (last status: %s)", err, lastStatus)
+			}
+			if lastReason == string(automotivev1alpha1.WorkspaceReasonContainerRestarting) {
+				handleError(caibcommon.NewActionableError(err, "caib workspace logs "+name))
+				return
+			}
+			handleError(fmt.Errorf("%w; contact your service administrator with this status", err))
```
Comment (maboras-rh) [inline cmd/caib/workspace/workspace.go]: Only `ContainerRestarting` gets specific actionable advice on timeout. IMO - extending this for other common timeout reasons like:
- `ImagePulling → verify the workspace image name and registry access`
- `Scheduling → check cluster capacity for the requested architecture`

These are user-actionable and more helpful than `contact your service administrator`

File: internal/buildapi/workspace_test.go
Comment (maboras-rh) [inline internal/buildapi/workspace_test.go]: missing test for `workspaceResponseFromCR` reason/message mapping
There is no test for this mapping or the clearing behavior. A test that creates a Workspace CR with `Status.Reason` and `Status.Message` set, then calls `workspaceResponseFromCR`, would catch regressions, especially for the transition edge cases.

File: internal/buildapi/workspace.go [hunk 2]
```diff
@@ -1012,6 +1014,10 @@ func buildWorkspaceResources(cpu, memory string, wsConfig *automotivev1alpha1.Wo
 		if err != nil {
 			return nil, fmt.Errorf("invalid --memory value %q: %v", memory, err)
 		}
+		minMemory := resource.MustParse("6Mi")
+		if q.Cmp(minMemory) < 0 {
+			return nil, fmt.Errorf("invalid --memory value %q: must be at least %s", memory, minMemory.String())
+		}
```
Comment (maboras-rh) [inline internal/buildapi/workspace.go]: the limit `6Mi` is validated in `buildWorkspaceResources` (in API/CLI layer), but not on the CRD itself. Users who create Workspace CRs directly via kubectl bypass this check. The controller will handle the resulting pod failure correctly - that's fine, but a kubebuilder validation would catch it earlier. 
nothing todo, It's just worth noting

File: internal/controller/workspace/controller.go [hunk 3]
```diff
@@ -135,6 +125,118 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
...
+	case "CreateContainerConfigError":
+		return automotivev1alpha1.WorkspaceReasonContainerConfigError, true
+	case "CreateContainerError", "RunContainerError":
+		return automotivev1alpha1.WorkspaceReasonContainerRuntimeError, true
+	case "ErrImageNeverPull", "InvalidImageName":
+		return automotivev1alpha1.WorkspaceReasonImageConfigError, true
+	case "ErrImagePull", "ImagePullBackOff":
+		return automotivev1alpha1.WorkspaceReasonImagePulling, false
+	case "CrashLoopBackOff":
+		return automotivev1alpha1.WorkspaceReasonContainerRestarting, false
```
Comment (maboras-rh) [inline internal/controller/workspace/controller.go]: `CrashLoopBackOff` is treated as non-fatal, so the workspace stays in "Creating" and the user gets no explicit failure signal until the CLI's 5-minute timeout. In practice, CrashLoopBackOff usually indicates a persistent issue (wrong entrypoint, missing binary, OOM on startup). is that intentional ? 
may consider tracking the restart count `status.RestartCount` and transitioning to `Failed` after a threshold, that would give faster feedback.

**Review by maboras-rh — APPROVED** [review]

File: internal/buildapi/workspace.go (hunk 2 above)
Comment (bennyz) [reply internal/buildapi/workspace.go]: users do not interact with kube api directly, everything goes through buildapi

File: cmd/caib/workspace/workspace.go (hunk 1 above)
Comment (bennyz) [reply cmd/caib/workspace/workspace.go]: i will think about improvements here, but in general in most cases users cannot access the cluster and know this

File: internal/buildapi/workspace.go (hunk 2 above)
Comment (maboras-rh) [reply internal/buildapi/workspace.go]: that's clear :-)

File: internal/controller/workspace/controller.go (hunk 3 above)
Comment (bennyz) [reply internal/controller/workspace/controller.go]: while CrashLookBackOff is usually terminal, by k8s it isn't, i've change it to display the phase during tries so it is visible before the timeout

File: internal/buildapi/workspace_test.go
Comment (bennyz) [reply internal/buildapi/workspace_test.go]: ack added

**Review by maboras-rh — APPROVED** [review]
